File: src/main/java/com/aladdinworksfourfourfour/service/NetworkEquipmentService.java
package com.aladdinworksfourfourfour.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworksfourfourfour.domain.NetworkEquipment;
import com.aladdinworksfourfourfour.dto.NetworkEquipmentDTO;
import com.aladdinworksfourfourfour.dto.NetworkEquipmentSearchDTO;
import com.aladdinworksfourfourfour.dto.NetworkEquipmentPageDTO;
import com.aladdinworksfourfourfour.dto.NetworkEquipmentConvertCriteriaDTO;
import com.aladdinworksfourfourfour.service.GenericService;
import com.aladdinworksfourfourfour.dto.common.RequestDTO;
import com.aladdinworksfourfourfour.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface NetworkEquipmentService extends GenericService<NetworkEquipment, Integer> {

	List<NetworkEquipment> findAll();

	ResultDTO addNetworkEquipment(NetworkEquipmentDTO networkEquipmentDTO, RequestDTO requestDTO);

	ResultDTO updateNetworkEquipment(NetworkEquipmentDTO networkEquipmentDTO, RequestDTO requestDTO);

    Page<NetworkEquipment> getAllNetworkEquipments(Pageable pageable);

    Page<NetworkEquipment> getAllNetworkEquipments(Specification<NetworkEquipment> spec, Pageable pageable);

	ResponseEntity<NetworkEquipmentPageDTO> getNetworkEquipments(NetworkEquipmentSearchDTO networkEquipmentSearchDTO);
	
	List<NetworkEquipmentDTO> convertNetworkEquipmentsToNetworkEquipmentDTOs(List<NetworkEquipment> networkEquipments, NetworkEquipmentConvertCriteriaDTO convertCriteria);

	NetworkEquipmentDTO getNetworkEquipmentDTOById(Integer networkEquipmentId);







}

File: aladdinworksfourfourfour_ui/src/services/NetworkEquipmentService.js
import http from "../http-common"; 

class NetworkEquipmentService {
  getAllNetworkEquipments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/networkEquipment/networkEquipments`, searchDTO);
  }

  get(networkEquipmentId) {
    return this.getRequest(`/networkEquipment/${networkEquipmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/networkEquipment?field=${matchData}`, null);
  }

  addNetworkEquipment(data) {
    return http.post("/networkEquipment/addNetworkEquipment", data);
  }

  update(data) {
  	return http.post("/networkEquipment/updateNetworkEquipment", data);
  }
  
  uploadImage(data,networkEquipmentId) {
  	return http.postForm("/networkEquipment/uploadImage/"+networkEquipmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new NetworkEquipmentService();
